app/models/generic_event.rb

```diff
@@ -95,4 +95,46 @@ def self.from_event(event)
 
     type.constantize.from_event(event)
   end
+
+  def self.details_attributes(*attributes)
+    define_singleton_method(:details_attributes) do
+      instance_variable_get('@details_attributes')
+    end
+
+    attributes = attributes.each_with_object([]) do |attribute_key, acc|
+      define_method(attribute_key) do
+        details[attribute_key]
+      end
+
+      define_method("#{attribute_key}=") do |attribute_value|
+        details[attribute_key] = attribute_value
+      end
+
+      acc << attribute_key
```

Comment:
I don't think you need this accumulator array, as it just ends up with the original contents of `attributes`.

Reply:
Removed :)